### ISSUE TITLE
Too large value used in testHitPattern

### ISSUE THREAD

**Issue by Dr15Jones** [cms-sw/cmssw#24309]
In the test DataFormats/TrackReco/test/testHitPattern.cpp, the test uses an array of uint32_t and is passing those values to a function that takes a uint16_t:
https://github.com/cms-sw/cmssw/blob/master/DataFormats/TrackReco/test/testHitPattern.cpp#L16

The values in the array are larger than the largest allowed 16bit number. 

This is generating a clang warning.

**Comment by Dr15Jones** [comment]
assign reconstruction

**Comment by Dr15Jones** [comment]
@VinInn FYI

**Comment by VinInn** [comment]
cool!
the test is supposed to test
```
bool appendHit(const DetId &id, TrackingRecHit::Type hitType, const TrackerTopology& ttopo);
```
that changed signature in the mean time (topology added)
NOT
```
bool appendHit(const uint16_t pattern, TrackingRecHit::Type hitType);
```

**Comment by slava77** [comment]
+1

this was apparently fixed in #25767
